tcl/auth.tcl

```diff
@@ -76,7 +76,7 @@ proc qc::auth_session { session_id } {
     #| Try to authenticate user based on the session_id given
     #| Return the user_id if successful
     #| On failure throw AUTH error
-    if { [session_exists $session_id] } {
+    if { [session_valid $session_id] } {
         return [session_user_id $session_id]
```

Comment:
`session_exists`
`session_valid`

Amazon have this model on their website.
A returning customer whom we recognise but we need to check if his session is valid before allowing more sensitive operations.

However the purpose of authentication is normally to check for permission to authorise the user to perform an action.
We could could call these "recognising" a user and "authenticating" as user.

The code as it stands in the lib doesn't make use of session_valid so doesn't go there with regards to validity.

May require `session_valid` to be used in `auth_session` rather than session_exists ?

Any ramifications for TLC checkout ?